pkg/controller/core/clusterqueue_controller.go

```diff
@@ -481,6 +568,41 @@ func (h *cqNamespaceHandler) Update(ctx context.Context, e event.UpdateEvent, _
 		}
 	}
 	h.qManager.QueueInadmissibleWorkloads(ctx, cqs)
+
+	if !features.Enabled(features.DefaultLocalQueue) {
```

Comment:
here I would prefer to wrap all the new logic under a new function to clearly show this is behind the feature gate